internal/route/api/v1/repo/contents.go

```diff
@@ -0,0 +1,209 @@
+	Encoding        string `json:"encoding,omitempty"`
+	Size            int64  `json:"size"`
+	Name            string `json:"name"`
+	Path            string `json:"path"`
+	Content         string `json:"content,omitempty"`
+	Sha             string `json:"sha"`
+	URL             string `json:"url"`
+	GitURL          string `json:"git_url"`
+	HTMLURL         string `json:"html_url"`
+	DownloadURL     string `json:"download_url"`
+	Links           Links  `json:"_links"`
+}
+
+type Links struct {
+	Git  string `json:"git"`
+	Self string `json:"self"`
+	HTML string `json:"html"`
+}
+
+func GetContents(c *context.APIContext) {
+	treeEntry, err := c.Repo.Commit.GetTreeEntryByPath(c.Repo.TreePath)
+	if err != nil {
+		c.NotFoundOrServerError("GetTreeEntryByPath", git.IsErrNotExist, err)
+		return
+	}
+
+	// TODO: figure out the best way to do this
+	// :base-url/:username/:project/raw/:refs/:path
+	templateDownloadURL := "%s/%s/%s/raw/%s"
+	// :base-url/repos/:username/:project/contents/:path
+	templateSelfLink := "%s/repos/%s/%s/contents/%s"
+	// :baseurl/repos/:username/:project/git/trees/:sha
+	templateGitURLLink := "%s/repos/%s/%s/trees/%s"
+	// :baseurl/repos/:username/:project/tree/:sha
+	templateHTMLLLink := "%s/repos/%s/%s/tree/%s"
+
+	gitURL := fmt.Sprintf(templateGitURLLink, c.BaseURL, c.Params(":username"), c.Params(":reponame"), treeEntry.ID.String())
+	htmlURL := fmt.Sprintf(templateHTMLLLink, c.BaseURL, c.Params(":username"), c.Params(":reponame"), treeEntry.ID.String())
+	selfURL := fmt.Sprintf(templateSelfLink, c.BaseURL, c.Params(":username"), c.Params(":reponame"), c.Repo.TreePath)
+
+	gr := &repoContents{
+		DownloadURL: fmt.Sprintf(templateDownloadURL, c.BaseURL, c.Params(":username"), c.Params(":reponame"), c.Repo.TreePath),
+		Size:        treeEntry.Size(),
+		Name:        treeEntry.Name(),
+		Path:        c.Repo.TreePath,
+		Sha:         treeEntry.ID.String(),
+		Links: Links{
+			Git:  gitURL,
+			Self: selfURL,
+			HTML: htmlURL,
+		},
+		URL:     selfURL,
+		GitURL:  gitURL,
+		HTMLURL: htmlURL,
+	}
+	// A tree entry can only be one of the following types:
+	// 1. Tree (Directory)
+	// 2. SubModule
+	// 3. SymLink
+	// 4. Blob
+	if treeEntry.IsSubModule() {
+		gr.Type = "submodule"
+		parsedURL, err := url.Parse(c.BaseURL)
+		if err != nil {
+			c.ServerError("ErrorURLParse", err)
+		}
+		host := parsedURL.Host
+		submoduleURL := fmt.Sprintf("git://%s/%s/%s.git", host, c.Params(":name"), c.Params(":reponame"))
+		gr.SubmoduleGitURL = submoduleURL
+		c.JSONSuccess(gr)
+		return
+
+	} else if treeEntry.IsLink() {
+		gr.Type = "symlink"
+		gr.Target = c.Repo.TreePath
+		c.JSONSuccess(gr)
+		return
+
+	} else if gr.Type == "blob" { // tree entry is a blob
+		gr.Type = "blob"
+		b, err := getBase64EncodedBlob(c)
+
+		if err != nil {
+			c.ServerError("GetBlobContent", err)
+			return
+		}
+
+		gr.Content = b
+		c.JSONSuccess(gr)
+		return
+	} else { // treeEntry is a directory
```

Comment:
This else case (and its indentation) is actually not needed. We can just do:
```suggestion
	...
	}
	
	// treeEntry is a directory
```

Then we can merge body of `AppendDirTreeEntries` function back to this function.

Reply:
merged the func back, it has removed those duplicate vars that I was using more than once in a couple of places. Thanks!